include/swift/AST/Decl.h

```diff
@@ -4874,8 +4874,14 @@ class ProtocolDecl final : public NominalTypeDecl {
   bool requiresSelfConformanceWitnessTable() const;
 
   /// Determine whether an existential type must be explicitly prefixed
-  /// with \c any. \c any is required if any of the members contain
-  /// an associated type, or if \c Self appears in non-covariant position.
+  /// with \c any. \c any is required if one of the following conditions is met
+  /// for this protocol or an inherited protocol:
+  /// - The protocol has an associated type requirement.
+  /// - `Self` appears in non-covariant position in the type signature of a
+  ///   value requirement.
+  ///
+  /// @Note This method does not take the state of language features into
+  /// account.
```

Comment:
You could also pass in the usage context into `existentialRequiresAny` so the method itself accounts for language feature state, but the `ExistentialRequiresAny` request does not. Thoughts?

Reply:
I was going to but that is what the serializer calls to obtain the evaluation result. I think we should rename the request to something like `HasSelfOrAssociatedTypeRequirements` and pair it with a corresponding method to use in serialization so that `existentialRequiresAny` could account for language feature state.

Reply:
👍 that sounds like a good approach to me (and we don't need to do that as part of this PR)